### PR TITLE
[Snowflake] Add support for inline columns in CTEs

### PR DESCRIPTION
**Pull request by vil1**

### CODE REVIEW

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/BundleInlineColumnExpressions.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.databricks.labs.remorph.parsers.snowflake.rules
+
+import com.databricks.labs.remorph.intermediate._
+
+class BundleInlineColumnExpressions extends Rule[LogicalPlan] {
+
+  private val inlineBundleTable = "INLINE_BUNDLE"
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/BundleInlineColumnExpressions.scala:7]: there might be many different names in CTE. Add a new tree node to wrap this name and change the names before they get to codegen.

File: core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala [hunk 2]
```diff
@@ -346,4 +346,27 @@ class SnowflakeToDatabricksTranspilerTest extends AnyWordSpec with TranspilerTes
...
+      """WITH
+        |    INLINE_BUNDLE (a, b, c) AS (SELECT * FROM VALUES (1, 2, 3)),
+        |    t (d, e) AS (SELECT 4, 5)
+        |SELECT
+        |    INLINE_BUNDLE.a + INLINE_BUNDLE.b,
+        |    INLINE_BUNDLE.a * INLINE_BUNDLE.c,
+        |    INLINE_BUNDLE.a * t.d
+        |FROM
+        |    INLINE_BUNDLE,
+        |    t;""".stripMargin
```
Comment (nfx) [inline core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala:369]: what about this?
```suggestion
      """WITH
        |    t (d, e) AS (SELECT 4, 5)
        |SELECT
        |    1 + 2,
        |    1 * 2,
        |    1 * t.d
        |FROM
        |    INLINE_BUNDLE,
        |    t;""".stripMargin
```

**Review by nfx — CHANGES_REQUESTED** [review]

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/BundleInlineColumnExpressions.scala (hunk 1 above)
Comment (nfx) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/BundleInlineColumnExpressions.scala:7]: and apparently, we don't need this anymore :)

File: core/src/main/scala/com/databricks/labs/remorph/intermediate/extensions.scala [hunk 3]
```diff
@@ -234,11 +234,25 @@ case object WEEK_INTERVAL extends KnownIntervalType
...
+
+// SNOWFLAKE - Common Table Expression may be of the form:
+// WITH
+//    a AS (1),
+//    b AS (2)
+// SELECT ...
+// CTEs like `a AS (1)` above act as columns of an anonymous table with a single row.
+// In AstBuilding phase, we translate those as InlineColumnExpression. Later, in Optimizing phase, we'll combine all
+// such expressions in a single table declaration (using VALUES).
+case class InlineColumnExpression(columnName: Id, value: Expression) extends LogicalPlan {
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/intermediate/extensions.scala:254]:
```suggestion
private[snowflake] case class InlineColumnExpression(columnName: Id, value: Expression) extends LogicalPlan {
```
and more it to the snowflake folder.

**Review by nfx — CHANGES_REQUESTED** [review]

**Review by sundarshankar89 — APPROVED** [review]
LGTM

File: core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala [hunk 4]
```diff
@@ -351,4 +351,25 @@ class SnowflakeToDatabricksTranspilerTest extends AnyWordSpec with TranspilerTes
...
+      |    c AS (3)
+      |SELECT
+      |    a + b,
+      |    a * c,
+      |    a * t.d
+      |FROM t;""".stripMargin transpilesTo
+      """WITH
+        |    t (d, e) AS (SELECT 4, 5)
+        |SELECT
+        |    1 + 2,
```
Comment (nfx) [inline core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala:368]: nice

**Review by nfx — APPROVED** [review]

**Pull request merged**